### ISSUE TITLE
[macOS] - Compilation failed - sprintf is depricated

### ISSUE THREAD

**Issue by bferguson3** [BambooTracker/BambooTracker#510]
> #### Checklist

> - [X] I am reporting exactly 1 bug with this issue.
> - [X] This bug hasn't already been reported.
> - [X] This bug hasn't already been fixed in the latest development build.

---

## Bug Description

On Sonoma 14.3.1 (presumably gcc and clang are all up to date) I get the following error (not warning) with compilation: 
```
chip/ymfm/ymfm_opn.cpp:442:10: error: 'sprintf' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Werror,-Wdeprecated-declarations]
                end += sprintf(end, " lfo=%02X", lfo_rate());
                       ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdio.h:180:1: note: 'sprintf' has been explicitly marked deprecated here
__deprecated_msg("This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead.")
^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/sys/cdefs.h:215:48: note: expanded from macro '__deprecated_msg'
        #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
```
## How to reproduce

make qmake_all & make

## System Information

- **Operating System**:  Sonoma 14.3.1, Apple Silicon M1
- **BambooTracker Version**:  Latest commit 
- **Build Type**: Pulled from github, run using make on cli

**Comment by bferguson3** [comment]
Doing a find and replace from 
`sprintf(end, `
to 
`snprintf(end, 256, '
fixed this issue. 

MacOS still needs other tweaks however -- as noted in source utils.hpp, result_of is deprecated, and needs to be changed to invoke_result. This breaks the ::type identifier and transformIf etc. etc.

**Comment by OPNA2608** [comment]
> as noted in source utils.hpp, result_of is deprecated, and needs to be changed to invoke_result.

Are you building with Qt6? Then this part likely falls under #506.

> By removing all instances of `-Werror` the application built successfully.

Try #508 + `CONFIG+=no_warnings_are_errors` at QMake configure time to work around these issues for now. If you're using CMake, `-DWARNINGS_ARE_ERRORS=OFF` will also accomplish this already.

> The distributed binary from the website does not launch with an unknown error, that is why I am building from source.

Can you give any details about this? I think this is either
- lack of an app signature - but I don't see how we could handle this properly (not ad-hoc) without getting apple hardware & signing up with apple to get a dev id, and moving the macOS CI to said hardware so the ID doesn't get put onto public infrastructure
- a broken app signature, due to implicitly-done ad-hoc signage + modifying the app contents during the packaging step - we could regenerate the ad-hoc signature after doing the changes in CI I guess
- some issue with Rosetta 2 + the used Qt version, or
- some new fancy Apple thing, iunno

If you're on the Discord server, feel free to ping me for debugging the issue with the binary. Might be faster than doing this here.

**Comment by bferguson3** [comment]
Closing as fixed per #508
